docs/topics/multiplatform/multiplatform-connect-to-apis.md

```diff
@@ -1,380 +1,276 @@
-            }
-        )
-    }
-    actual fun closeSocket(code: Int, reason: String) {
-        webSocket?.close(code, reason)
-        webSocket = null
-    }
-    actual fun sendMessage(msg: String) {
-        webSocket?.send(msg)
-    }
+// In the iosMain source set:
+class IOSPlatform : Platform
+
+actual fun platform() = IOSPlatform()
+```
+
+You call `platform()` in the common code and can work with an object of the `Platform` type. When you run this common
+code on Android, the `platform()` call will return an instance of the `AndroidPlatform` class. When you run it on
+iOS, `platform()` will return an instance of the `IOSPlatform` class.
+
+### Different entry points
+
+If you control the entry points, you can construct implementations of each platform artifact without using the
+expect/actual mechanism. This way, you define the platform implementations in the shared Kotlin Multiplatform module,
+but instantiate them in platform modules:
+
+```kotlin
+// Shared Kotlin Multiplatform module
+// In the commonMain source set:
+interface Platform
+
+fun application(p: Platform) {
+    // application logic
 }
 ```
 
-Android implementation uses the third-party library [OkHttp](https://square.github.io/okhttp/). Add the corresponding dependency to `build.gradle(.kts)` in the shared module:
+```kotlin
+// In the androidMain source set:
+class AndroidPlatform : Platform
+```
 
-<tabs group="build-script">
-<tab title="Kotlin" group-key="kotlin">
+```kotlin
+// In the iosMain source set:
+class IOSPlatform : Platform
+```
 
 ```kotlin
-sourceSets {
-    val androidMain by getting {
-        dependencies {
-            implementation("com.squareup.okhttp3:okhttp:$okhttp_version")
-        }
+// In the androidApp platform module:
+import android.app.Application
+import mysharedpackage.*
+
+class MyApp : Application() {
+    override fun onCreate() {
+        super.onCreate()
+        application(AndroidPlatform())
     }
 }
 ```
 
-</tab>
-<tab title="Groovy" group-key="groovy">
+```Swift
+// In the iosApp platform module (in Swift):
+import shared
 
-```groovy
-commonMain {
-    dependencies {
-        implementation "com.squareup.okhttp3:okhttp:$okhttp_version"
+@main
+struct iOSApp : App {
+    init() {
+        application(IOSPlatform())
     }
 }
 ```
 
-</tab>
-</tabs>
+On Android, you create an instance of `AndroidPlatform` and pass it to the `application()` function, while on iOS, you
+similarly create and pass an instance of `IOSPlatform`. These entry points don't need to be the entry points of your
+apps, but this is where you can call the specific functionality of your shared module.
+
+Providing correct implementations with expected and actual functions or directly through entry points works well for
+simple scenarios. However, if you use DI in your project, we recommend you use it in such cases as well for consistency.
```

Comment:
Since DI is discussed in the next section, should we remove this paragraph to avoid duplication of content?